### PR TITLE
Handle missing file error in fmt more gracefully

### LINKED ISSUE #5363: Return a regular error, not exception for missing file in `fmt`
```
$ lfortran --fixed-form fmt
integration_tests/functions_24.f90
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/home/ondrej/repos/lfortran/lfortran/src/bin/lfortran", in _start
  Binary file "/lib/x86_64-linux-gnu/libc.so.6", in __libc_start_main
  Binary file "/lib/x86_64-linux-gnu/libc.so.6", in <invalid>
  File "/home/ondrej/repos/lfortran/lfortran/src/bin/lfortran.cpp", line 2686, in main
    return main_app(argc, argv);
LCompilersException: File does not exist: integration_tests/functions_24.f90
```

This should be fixed similarly to https://github.com/lfortran/lfortran/pull/5289.

### PR DESCRIPTION
**Pull request by mohitdhingra22**
Fixed #5363
The error handling for missing files in the fmt functionality of the LFortran compiler. Instead of throwing an exception that leads to an ‘Internal Compiler Error,’ it now provides a clear and user-friendly error message and exits gracefully.

### CODE REVIEW

**Comment by gxyd** [comment]
Please link the issue in the pull request description that this pull request fixed.

File: .vscode/launch.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
             "type": "cppdbg",
             "request": "launch",
             "program": "${workspaceFolder}/src/bin/lfortran",
-            "args": ["examples/expr2.f90"],
+            "args": ["--fixed-form","fmt","test2.f90"],
```
Comment (gxyd) [inline .vscode/launch.json:13]: Could you please undo this change? This isn't supposed to be committed.

File: .vscode/launch.json [hunk 2]
```diff
@@ -1,23 +0,0 @@
-{
```
[the remark this answers is not in the document]
Comment (mohitdhingra22) [reply .vscode/launch.json:1]: Any method to undo the deleted file?

[the remark this answers is not in the document]
Comment (gxyd) [reply .vscode/launch.json:1]: If you don't want to do a force-push in git, you can just create a `.vscode/launch.json` file with the same contents as the original file in a new commit, that's one way. (Make sure there no new whitespaces like newline or spaces though)

Another way is you use `git reset`. I can explain more in details if you want me to (that will require a force-push)

[the remark this answers is not in the document]
Comment (gxyd) [reply .vscode/launch.json:1]: Also, please make sure you create PR against the latest *main*, as it seems like your branch doesn't have the latest changes.

You can try merging *main* into your branch.

[the remark this answers is not in the document]
Comment (mohitdhingra22) [reply .vscode/launch.json:1]: got it!

[the remark this answers is not in the document]
Comment (Pranavchiku) [reply .vscode/launch.json:1]: Simply do:

```
git reset --soft HEAD^
git push <your-remote> <branch-name> -f
```

and then

```
git restore --staged .
git restore .
```

**Review by Pranavchiku — APPROVED** [review]
Looks good just revert the last commit and it shall be fine.

File: .vscode/launch.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (mohitdhingra22) [reply .vscode/launch.json:1]: > Simply do:
> 
> ```
> git reset --soft HEAD^
> git push <your-remote> <branch-name> -f
> ```
> 
> and then
> 
> ```
> git restore --staged .
> git restore .
> ```

I did these steps but nothing updated in PR.

[the remark this answers is not in the document]
Comment (Pranavchiku) [reply .vscode/launch.json:1]: Do a `git log` check if the commit [Delete .vscode/launch.json](https://github.com/lfortran/lfortran/pull/5401/commits/0708d2ca35a958646e0fd0427531cb88159fc502) exists, if not then do `git pull <your-remote> <your-branch>` and then recheck with `git log` and follow the steps.

[the remark this answers is not in the document]
Comment (mohitdhingra22) [reply .vscode/launch.json:1]:
```
(lf) mohitdhingra@Mohits-MacBook-Air lfortran % git log
commit 0708d2ca35a958646e0fd0427531cb88159fc502 (HEAD -> Fix1, origin/Fix1)
Author: mxyd <154071159+mohitdhingra22@users.noreply.github.com>
Date:   Sat Nov 16 15:10:07 2024 +0530

    Delete .vscode/launch.json
```
It exist!

[the remark this answers is not in the document]
Comment (Pranavchiku) [reply .vscode/launch.json:1]: Then it shall work ;)

**Comment by Pranavchiku** [comment]
There you go!

File: .vscode/launch.json (hunk 1 above)
Comment (Pranavchiku) [inline .vscode/launch.json:13]: Now do `git pull <your-branch> <your-remote>` and then add a new commit:
```suggestion
            "args": ["examples/expr2.f90"],
```

Comment (certik) [reply .vscode/launch.json:21]: @mohitdhingra22 go ahead and apply the above change.

Comment (mohitdhingra22) [reply .vscode/launch.json]: All done or any changes to do?

**Comment by mohitdhingra22** [comment]
I have resolved the issue @certik @Pranavchiku

**Review by gxyd — APPROVED** [review]
LGTM!
